src/java.base/share/classes/sun/nio/cs/StreamDecoder.java

```diff
@@ -271,6 +273,26 @@ private boolean isOpen() {
         return !closed;
     }
 
+    public void fillZeroToPosition() throws IOException {
+        Object lock = this.lock;
+        if (lock instanceof InternalLock locker) {
+            locker.lock();
+            try {
+                lockedFillZeroToPosition();
+            } finally {
+                locker.unlock();
+            }
+        } else {
+            synchronized (lock) {
+                lockedFillZeroToPosition();
+            }
+        }
+    }
+
+    private void lockedFillZeroToPosition() throws IOException {
+        ensureOpen();
```

Comment:
Hello Naoto, is this open check needed? Given that we are just zeroing out already read content, would it thus be OK to clear that content to zero even after this `Reader` instance has been closed?

Reply:
Thanks, Jai. Removed it.